streamer/src/nonblocking/quic.rs

```diff
@@ -514,10 +518,60 @@ async fn setup_connection(
     }
 }
 
+async fn patch_batch_sender(packet_sender: Sender<PacketBatch>, 
+    packet_receiver: AsyncReceiver<(Meta, Bytes, u64, usize)>, 
+    exit: Arc<AtomicBool>,
+    stats: Arc<StreamStats>,) {
+    loop {
+        let mut packet_batch = PacketBatch::with_capacity(64);
+        //todo: perhaps we should be timing from the first time the packet batch becomes non-zero length?
+        let last_sent = Instant::now();
+        loop {
+            if exit.load(Ordering::Relaxed) {
+                return;
+            }
+            if let Ok(recv_result) = tokio::time::timeout(
+                Duration::from_micros(250),
+                packet_receiver.recv(),
+            )
+            .await {
+                if let Ok((meta, bytes, offset, end_of_chunk)) = recv_result {
```

Comment:
I wonder if sender for this channel was dropped not due to exit signal (not sure if possible), than we will get error here over and over again and never exit from the loop.

Reply:
Yes, I did actually notice this. When the quic server is told to exit (which is not uncommon, as we spin up a gossip server and destroy it during initial discovery). Not much harm done since it that situation, we're tearing down everything anyway. But it does sometimes lead to some spurious errors in the logs. Might be a good to check the exit flag first before logging an error...